services/housetables/src/main/java/com/linkedin/openhouse/housetables/services/UserTablesServiceImpl.java

```diff
@@ -150,21 +155,53 @@ public void renameUserTable(
       // TODO: Use toDataBaseId for destination instead of fromDatabaseId once rename across
       // databases is supported
       htsJdbcRepository.renameTableId(
-          fromDatabaseId, fromTableId, fromDatabaseId, toTableId, metadataLocation);
+          fromDatabaseId, fromTableId, fromDatabaseId, toTableId, metadataLocation, false);
     } catch (DataIntegrityViolationException e) {
       throw new AlreadyExistsException("Table", toTableId);
     }
   }
 
   @Override
-  public void deleteUserTable(String databaseId, String tableId) {
-    if (!htsJdbcRepository.existsById(
-        UserTableRowPrimaryKey.builder().databaseId(databaseId).tableId(tableId).build())) {
-      throw new NoSuchUserTableException(databaseId, tableId);
+  public void deleteUserTable(String databaseId, String tableId, boolean isSoftDeleted) {
+    if (!isSoftDeleted) {
+      if (!htsJdbcRepository.existsById(
+          UserTableRowPrimaryKey.builder().databaseId(databaseId).tableId(tableId).build())) {
+        throw new NoSuchUserTableException(databaseId, tableId);
+      }
+      htsJdbcRepository.deleteById(
+          UserTableRowPrimaryKey.builder().databaseId(databaseId).tableId(tableId).build());
```

Comment:
I thought we said we would rename the tableId to something like tableId@<deletion_timestamp> Is that not happening as part of soft-delete?
We need to allow for recreation of table with same name when previous table was soft-deleted.